pkg/pod/pod.go

```diff
@@ -84,7 +84,6 @@ var (
 	}, {
 		Name:      "tekton-internal-steps",
 		MountPath: pipeline.StepsDir,
-		ReadOnly:  true,
```

Comment:
This is a big change though 🤔. This volume is readonly for a reason 🙃 
cc @wlynch

Reply:
Great catch @vdemeester. 

I suggested in another comment that we could make the step dir inside `/tekton/results` and write the results there `/tekton/results/stepName/resultFile`? Instead of making  `/tekton/steps` writeable and then doing `/tekton/steps/stepName/resultFile`?

Another nice thing is that all results are in the same location.
WDYT?

Reply:
For now I've taken this approach. i.e. writing step results in `/tekton/results/stepName/resultName`. This way, we are not mutating anything about `StepsDir`.  I can revert/updated it if needed.

Reply:
You want `/tekton/steps/$x/results`. This folder is writeable by its matching step, and readonly to every other step, so you shouldn't need to remove the readonly bit.

This was meant to be a speedbump to prevent steps from being able to overwrite other step data (it's not 100% perfect / you can easily privilege override it, but it's good enough for most use cases).

Reply:
Sounds good! I write to `/tekton/steps/stepName/results/`.